src/Microsoft.Azure.SignalR/Internals/DefaultHubMessageSerializer.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.AspNetCore.SignalR;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class DefaultHubMessageSerializer
```

Comment:
Is this the code you copied from SignalR server? Can you put a comment in saying which commit it's from, etc. (like, "Copied from [full GitHub URL]" or something). I'd like to be able to cross-reference this later.

In 5.0 we plan to look at introducing a more public "API" for this so we don't have to copy code around :).